src/test/java/bio/terra/service/snapshotbuilder/utils/ConceptChildrenQueryBuilderTest.java

```diff
@@ -15,6 +15,44 @@
 @Tag(Unit.TAG)
 class ConceptChildrenQueryBuilderTest {
 
+  private static final String GCP_EXPECTED =
+      """
+      SELECT c.concept_name, c.concept_id,
+        COUNT(DISTINCT c0.person_id) AS count,
+        EXISTS (SELECT 1
+          FROM concept_ancestor AS c1
+                   JOIN concept AS c2 ON c2.concept_id = c1.descendant_concept_id
+          WHERE (c1.ancestor_concept_id = c.concept_id AND
+                 c1.descendant_concept_id != c.concept_id AND c2.standard_concept = 'S'))
+        AS has_children
+      FROM concept AS c
+        JOIN concept_ancestor AS c3 ON c3.ancestor_concept_id = c.concept_id
+        JOIN condition_occurrence AS c0 ON c0.condition_concept_id = c3.descendant_concept_id
+      WHERE (c.concept_id IN (SELECT c4.concept_id_2
+          FROM concept_relationship AS c4
+          WHERE (c4.concept_id_1 = 101 AND c4.relationship_id = 'Subsumes')) AND c.standard_concept = 'S')
+      GROUP BY c.concept_name, c.concept_id
+      ORDER BY c.concept_name ASC""";
+
+  private static final String AZURE_EXPECTED =
+      """
+      SELECT c.concept_name, c.concept_id,
+        COUNT(DISTINCT c0.person_id) AS count,
+        CASE WHEN EXISTS (SELECT 1
+          FROM concept_ancestor AS c1
+                   JOIN concept AS c2 ON c2.concept_id = c1.descendant_concept_id
+          WHERE (c1.ancestor_concept_id = c.concept_id AND
+                 c1.descendant_concept_id != c.concept_id AND c2.standard_concept = 'S'))
+        THEN 1 ELSE 0 END AS has_children
+      FROM concept AS c
+        JOIN concept_ancestor AS c3 ON c3.ancestor_concept_id = c.concept_id
+        JOIN condition_occurrence AS c0 ON c0.condition_concept_id = c3.descendant_concept_id
+      WHERE (c.concept_id IN (SELECT c4.concept_id_2
+          FROM concept_relationship AS c4
+          WHERE (c4.concept_id_1 = 101 AND c4.relationship_id = 'Subsumes')) AND c.standard_concept = 'S')
+      GROUP BY c.concept_name, c.concept_id
+      ORDER BY c.concept_name ASC""";
```

Comment:
If the tests use the table name generator x -> x, as is specified in the QueryTestUtils, why do we need to represent cloud specific sql here? Can that be tested elsewhere?

Reply:
The cloud specific part is the `has_children` expression. I agree it'd be better to not test that here but we don't have a good way to do that now, as far as I know.